homeassistant/components/cloud/alexa_config.py

```diff
@@ -318,32 +322,34 @@ def async_invalidate_access_token(self) -> None:
 
     async def async_get_access_token(self) -> str | None:
         """Get an access token."""
+        details: AlexaAccessTokenDetails | None
         if self._token_valid is not None and self._token_valid > utcnow():
             return self._token
 
-        resp = await cloud_api.async_alexa_access_token(self._cloud)
-        body = await resp.json()
-
-        if resp.status == HTTPStatus.BAD_REQUEST:
-            if body["reason"] in ("RefreshTokenNotFound", "UnknownRegion"):
-                if self.should_report_state:
-                    persistent_notification.async_create(
-                        self.hass,
-                        (
-                            "There was an error reporting state to Alexa"
-                            f" ({body['reason']}). Please re-link your Alexa skill via"
-                            " the Alexa app to continue using it."
-                        ),
-                        "Alexa state reporting disabled",
-                        "cloud_alexa_report",
-                    )
-                raise alexa_errors.RequireRelink
+        try:
+            details = await self._cloud.alexa_api.access_token()
+        except AlexaApiNeedsRelinkError as exception:
+            if self.should_report_state:
+                persistent_notification.async_create(
```

Comment:
Ideally, this should be migrated to a repair issue.
But that is out of scope here.